#! /bin/sh /usr/share/dpatch/dpatch-run
## 10mscodec_null_strcmp.dpatch by Samuel Mimram <smimram@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Check that we are not using NULL pointers.

@DPATCH@
diff -urNad linphone-1.0.1/mediastreamer/mscodec.c /tmp/dpep.crLUDd/linphone-1.0.1/mediastreamer/mscodec.c
--- linphone-1.0.1/mediastreamer/mscodec.c	2005-03-30 11:14:09.000000000 +0200
+++ /tmp/dpep.crLUDd/linphone-1.0.1/mediastreamer/mscodec.c	2005-03-30 11:18:23.000000000 +0200
@@ -82,9 +82,9 @@
 	while (elem!=NULL && name!=NULL)
 	{
 		info=(MSFilterInfo *)elem->data;
-		if ( (info->type==MS_FILTER_AUDIO_CODEC) ){
+		if (info && (info->type==MS_FILTER_AUDIO_CODEC) ){
 			MSCodecInfo *codinfo=(MSCodecInfo *)info;
-			if (strcmp(codinfo->description,name)==0){
+			if (codinfo->description && strcmp(codinfo->description,name)==0){
 				return MS_CODEC_INFO(info);
 			}
 		}
@@ -100,9 +100,9 @@
 	while (elem!=NULL && name!=NULL)
 	{
 		info=(MSFilterInfo *)elem->data;
-		if ( (info->type==MS_FILTER_VIDEO_CODEC) ){
+		if (info && (info->type==MS_FILTER_VIDEO_CODEC) ){
 			MSCodecInfo *codinfo=(MSCodecInfo *)info;
-			if (strcmp(codinfo->description,name)==0){
+			if (codinfo->description && strcmp(codinfo->description,name)==0){
 				return MS_CODEC_INFO(info);
 			}
 		}
@@ -137,7 +137,7 @@
 		info=(MSFilterInfo *)elem->data;
 		if ((info->type==MS_FILTER_AUDIO_CODEC) || (info->type==MS_FILTER_VIDEO_CODEC)){
 			MSCodecInfo *codinfo=(MSCodecInfo *)elem->data;
-			if (strcmp(info->name,name)==0){
+			if (info->name && strcmp(info->name,name)==0){
 				return codinfo->encoder();
 			}
 		}
@@ -155,7 +155,7 @@
 		info=(MSFilterInfo *)elem->data;
 		if ((info->type==MS_FILTER_AUDIO_CODEC) || (info->type==MS_FILTER_VIDEO_CODEC)){
 			MSCodecInfo *codinfo=(MSCodecInfo *)elem->data;
-			if (strcmp(info->name,name)==0){
+			if (info->name && strcmp(info->name,name)==0){
 				return codinfo->decoder();
 			}
 		}
